src/task_queue/database_tasks/backfill_data.py

```diff
@@ -0,0 +1,343 @@
+These were formerly in a 'data.backfill' route in misc.py.  
+Some are obsolete and not relevant any longer, but kept for posterity in case needed.
+
+1. backfill_subdomains: loop over communities, created Subdomain instances for each
+2. backfill_teams: loop through Teams, create TeamMember instances
+3. backfill_community_memebers: loop through UserProfile and CommunityAdminGroups to create CommunityJembers
+4. backfill_graph_default_data: fix action done category data
+5. backfill_locations: add more complete address fields to RealEstateUnits and Communities
+"""
+import csv
+import datetime
+from sentry_sdk import capture_message
+import zipcodes
+from django.http import HttpResponse
+from database.models import (
+    Subdomain,
+    Community,
+    Team,
+    TeamMember,
+    CommunityMember,
+    RealEstateUnit,
+    CommunityAdminGroup,
+    UserProfile,
+    Data,
+    TagCollection,
+    UserActionRel,
+    Location,
+)
+from database.utils.common import json_loader
+from _main_.utils.emailer.send_email import send_massenergize_email_with_attachments
+from api.store.utils import find_reu_community, split_location_string, check_location
+from api.utils.constants import DATA_DOWNLOAD_TEMPLATE
+
+
+def write_to_csv(data):
+    response = HttpResponse(content_type="text/csv")
+    writer = csv.DictWriter(response, fieldnames=["Message", "Error"])
+    writer.writeheader()
+    for row in data:
+        writer.writerow(row)
+    return response.content
+
+
+def backfill_data(task=None):
+    try:
+        #data = backfill_subdomains()
+        #data = backfill_teams()
+        #data = backfill_community_members()
+        #data = backfill_graph_default_data()
+        data = backfill_locations()
+        if len(data) > 0:
+            report =  write_to_csv(data)
+            temp_data = {'data_type': "Content Spacing", "name":task.creator.full_name if task.creator else "admin"}
+            file_name = "Data-Backfill-Report-{}.csv".format(datetime.datetime.now().strftime("%Y-%m-%d"))
+            send_massenergize_email_with_attachments(DATA_DOWNLOAD_TEMPLATE,temp_data,[task.creator.email], report, file_name)
+        return True
+    
+    except Exception as e:
+        print(str(e))
+        capture_message(str(e), level="error")
+        return False
+
+
+def backfill_subdomans():
+    data = []
+    for c in Community.objects.all():
+        try:
+            data.append({"Message": "Creating "+c.subdomain, "Error":""})
+            Subdomain(name=c.subdomain, in_use=True, community=c).save()
+        except Exception as e:
+            data.append({"Message": "Error", "Error": str(e)})
+    return data
+
+
+def backfill_teams():
+    data = []
+    try:
+        teams = Team.objects.all()
+        for team in teams:
+            members = team.members.all()
+            for member in members:
+                team_member = TeamMember.objects.filter(
```

Comment:
This is old code which I haven't changed, just moved it from another place; it is there in case we need it.
The reason for this routine was that the TeamMember model was introduced at some point for some purpose, and needed to be filled in using the team.members.  So team.members is redundant, and could probably be eliminated but it hasn't been a priority